tests/__init__.py

```diff
@@ -242,3 +247,74 @@ def cmd(self, *cmd):
         if result != b'OK':
             raise RuntimeError(
                 "Error from command '%s': %s" % (cmd, result))
+
+
+# This is added to this file because it's used in both
+# the functional and unit tests for cred refresh.
+class IntegerRefresher(credentials.RefreshableCredentials):
+    """Refreshable credentials to help with testing.
+
+    This class makes testing refreshable credentials easier.
+    It has the following functionality:
+
+        * A counter, self.refresh_counter, to indicate how many
+          times refresh was called.
+        * A way to specify how many seconds to make credentials
+          valid.
+        * Configurable advisory/mandatory refresh.
+        * An easy way to check consistency.  Each time creds are
+          refreshed, all the cred values are set to the next
+          incrementing integer.  Frozen credentials should always
+          have this value.
+    """
+
+    _advisory_refresh_timeout = 2
+    _mandatory_refresh_timeout = 1
+    _credentials_expire = 3
+
+    def __init__(self, creds_last_for=_credentials_expire,
+                 advisory_refresh=_advisory_refresh_timeout,
+                 mandatory_refresh=_mandatory_refresh_timeout,
+                 refresh_function=None):
+        expires_in = (
+            self._current_datetime() +
+            datetime.timedelta(seconds=creds_last_for))
+        if refresh_function is None:
+            refresh_function = self._do_refresh
+        super(IntegerRefresher, self).__init__(
+            '1', '1', '1', expires_in,
```

Comment:
Is their a particular reason you picked ``1`` as the starting point? It is a little confusing in the unit test that you assert that the value is ``2`` which I would see as the credentials being successfully refreshed twice when you are making sure that they get immediately refreshed. Maybe start at 0? Or have a helper function for asserting how many times credentials were refreshed?